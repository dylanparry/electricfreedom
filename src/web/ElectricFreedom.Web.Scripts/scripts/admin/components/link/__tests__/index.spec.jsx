import React from 'react';
import { RouterStore, Route } from 'mobx-router';
import { shallow } from 'enzyme';

import Link from '../';

const routes = {
  basic: new Route({
    path: '/test',
    component: null,
  }),
  withParams: new Route({
    path: '/test/:param',
    component: null,
  }),
};

const setup = (propOverrides) =>
{
  const props = Object.assign({
    children: 'Test Link',
    view: routes.basic,
    params: {},
    store: {
      router: new RouterStore(),
    },
  }, propOverrides);

  const wrapper = shallow(<Link {...props}>{props.children}</Link>);

  return {
    wrapper,
    text: wrapper.dive().dive().text(), // The link text
    href: wrapper.dive().dive().prop('href'), // The link href
  };
};

describe('Link Component', () =>
{
  test('render', () =>
  {
    const { wrapper } = setup();

    expect(wrapper).toMatchSnapshot();
  });

  test('with children', () =>
  {
    const { text } = setup({
      children: 'Other Text',
    });

    expect(text).toBe('Other Text');
  });

  test('with a "view" attribute', () =>
  {
    const { href } = setup({
      view: routes.basic,
    });

    expect(href).toBe('/test');
  });

  test('with a "params" attribute', () =>
  {
    const { href } = setup({
      view: routes.withParams,
      params: { param: 123 },
    });

    expect(href).toBe('/test/123');
  });
});
