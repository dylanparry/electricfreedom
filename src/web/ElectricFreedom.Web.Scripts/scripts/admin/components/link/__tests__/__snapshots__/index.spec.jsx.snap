// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link Component render 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <link
    params={Object {}}
    store={
        Object {
            "router": RouterStore {
              "goTo": [Function],
            },
          }
    }
    view={
        Route {
            "component": null,
            "getParamsObject": [Function],
            "getRootPath": [Function],
            "goTo": [Function],
            "originalPath": "/test",
            "path": "/test",
            "replaceUrlParams": [Function],
            "rootPath": "/test",
          }
    }
>
    Test Link
</link>,
  "nodes": Array [
    <link
      params={Object {}}
      store={
            Object {
                  "router": RouterStore {
                    "goTo": [Function],
                  },
                }
      }
      view={
            Route {
                  "component": null,
                  "getParamsObject": [Function],
                  "getRootPath": [Function],
                  "goTo": [Function],
                  "originalPath": "/test",
                  "path": "/test",
                  "replaceUrlParams": [Function],
                  "rootPath": "/test",
                }
      }
>
      Test Link
</link>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <inject-link-with-store
        params={Object {}}
        store={
                Object {
                        "router": RouterStore {
                          "goTo": [Function],
                        },
                      }
        }
        view={
                Route {
                        "component": null,
                        "getParamsObject": [Function],
                        "getRootPath": [Function],
                        "goTo": [Function],
                        "originalPath": "/test",
                        "path": "/test",
                        "replaceUrlParams": [Function],
                        "rootPath": "/test",
                      }
        }
>
        Test Link
</inject-link-with-store>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance":  {
        "_reactInternalInstance": [Circular],
        "context": Object {
          "mobxStores": undefined,
        },
        "props": Object {
          "children": "Test Link",
          "params": Object {},
          "store": Object {
            "router": RouterStore {
              "goTo": [Function],
            },
          },
          "view": Route {
            "component": null,
            "getParamsObject": [Function],
            "getRootPath": [Function],
            "goTo": [Function],
            "originalPath": "/test",
            "path": "/test",
            "replaceUrlParams": [Function],
            "rootPath": "/test",
          },
        },
        "refs": Object {},
        "state": null,
        "storeRef": [Function],
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <link
          params={Object {}}
          store={
                    Object {
                              "router": RouterStore {
                                "goTo": [Function],
                              },
                            }
          }
          view={
                    Route {
                              "component": null,
                              "getParamsObject": [Function],
                              "getRootPath": [Function],
                              "goTo": [Function],
                              "originalPath": "/test",
                              "path": "/test",
                              "replaceUrlParams": [Function],
                              "rootPath": "/test",
                            }
          }
>
          Test Link
</link>,
        "_debugID": 2,
        "_renderedOutput": <link
          params={Object {}}
          store={
                    Object {
                              "router": RouterStore {
                                "goTo": [Function],
                              },
                            }
          }
          view={
                    Route {
                              "component": null,
                              "getParamsObject": [Function],
                              "getRootPath": [Function],
                              "goTo": [Function],
                              "originalPath": "/test",
                              "path": "/test",
                              "replaceUrlParams": [Function],
                              "rootPath": "/test",
                            }
          }
>
          Test Link
</link>,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <inject-link-with-store
    params={Object {}}
    store={
        Object {
            "router": RouterStore {
              "goTo": [Function],
            },
          }
    }
    view={
        Route {
            "component": null,
            "getParamsObject": [Function],
            "getRootPath": [Function],
            "goTo": [Function],
            "originalPath": "/test",
            "path": "/test",
            "replaceUrlParams": [Function],
            "rootPath": "/test",
          }
    }
>
    Test Link
</inject-link-with-store>,
}
`;
